packages/react/src/components/NumberInput/NumberInput.js

```diff
@@ -113,11 +113,11 @@ class NumberInput extends Component {
     /**
      * Optional starting value for uncontrolled state
      */
-    defaultValue: PropTypes.number,
+    defaultValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
     /**
      * Specify the value of the input
      */
-    value: PropTypes.number,
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```

Comment:
Should this also allow `null`? 🤔 
I ask because when experimenting in storybook... if you delete `value`, then in the React components plugin `value=null`. Seems reasonable to expect someone to *remove* `value` in this case?